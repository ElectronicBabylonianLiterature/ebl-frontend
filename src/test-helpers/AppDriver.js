import React from 'react'
import { fireEvent, render, waitForElement } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import App from 'App'
import Auth from 'auth/Auth'
import WordRepository from 'dictionary/WordRepository'
import FragmentRepository from 'fragmentarium/FragmentRepository'
import ImageRepository from 'fragmentarium/ImageRepository'
import FragmentService from 'fragmentarium/FragmentService'
import WordService from 'dictionary/WordService'
import TextService from 'corpus/TextService'
import Session from 'auth/Session'
import BibliographyRepository from 'bibliography/BibliographyRepository'
import BibliographyService from 'bibliography/BibliographyService'
import { defaultErrorReporter } from 'ErrorReporterContext'
import createAuth0Config from 'auth/createAuth0Config'

function createApp(api, sessionStore) {
  const auth0Config = createAuth0Config()
  const auth = new Auth(sessionStore, defaultErrorReporter, auth0Config)
  const wordRepository = new WordRepository(api)
  const fragmentRepository = new FragmentRepository(api)
  const imageRepository = new ImageRepository(api)
  const bibliographyRepository = new BibliographyRepository(api)
  const wordService = new WordService(wordRepository)
  const bibliographyService = new BibliographyService(bibliographyRepository)
  const fragmentService = new FragmentService(
    fragmentRepository,
    imageRepository,
    bibliographyService
  )
  const textService = new TextService(api)

  return (
    <App
      auth={auth}
      wordService={wordService}
      fragmentService={fragmentService}
      bibliographyService={bibliographyService}
      textService={textService}
    />
  )
}

export default class AppDriver {
  initialEntries = []
  element = null
  session = null

  constructor(api) {
    this.api = api
  }

  withPath(path) {
    this.initialEntries = [path]
    return this
  }

  withSession() {
    this.session = new Session(
      'accessToken',
      'idToken',
      Number.MAX_SAFE_INTEGER,
      ['write:texts']
    )
    return this
  }

  render() {
    const fakeSessionStore = {
      setSession: session => {
        this.session = session
      },
      clearSession: () => {
        this.session = null
      },
      getSession: () => this.session || new Session('', '', 0, [])
    }

    this.element = render(
      <MemoryRouter initialEntries={this.initialEntries}>
        {createApp(this.api, fakeSessionStore)}
      </MemoryRouter>
    )

    return this
  }

  async waitForText(text) {
    await waitForElement(() => this.element.getByText(text))
  }

  expectTextContent(text) {
    expect(this.element.container).toHaveTextContent(text)
  }

  expectNotInContent(text) {
    expect(this.element.queryByText(text)).toBeNull()
  }

  expectLink(text, expectedHref) {
    expect(this.element.getByText(text)).toHaveAttribute('href', expectedHref)
  }

  expectBreadcrumbs(crumbs) {
    this.expectTextContent(crumbs.join(''))
  }

  expectInputElement(label, expectedValue) {
    expect(this.element.getByLabelText(label).value).toEqual(
      String(expectedValue)
    )
  }

  changeValueByLabel(label, newValue) {
    const input = this.element.getByLabelText(label)
    fireEvent.change(input, { target: { value: newValue } })
  }

  click(text, n = 0) {
    const clickable = this.element.getAllByText(text)[n]
    fireEvent.click(clickable)
    // await wait()
  }
}
