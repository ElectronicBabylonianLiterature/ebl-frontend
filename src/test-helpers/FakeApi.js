import Promise from 'bluebird'

export default class FakeApi {
  gets = []
  client = {
    fetchJson: (path, authenticate) => {
      const expectation = this.gets.find(entry => entry.path === path && entry.authenticate === authenticate)
      return expectation
        ? Promise.resolve(expectation.response)
        : Promise.reject(new Error(`Unexpected ${authenticate ? 'authenticated' : 'not-authenticated'} fetchJson: ${path}`))
    },

    fetchBlob: (path, authenticate) => {
      return Promise.reject(new Error(`Unexpected ${authenticate ? 'authenticated' : 'not-authenticated'} fetchBlob: ${path}`))
    }
  }

  expectText (text) {
    this.gets.push({
      path: `/texts/${text.category}/${text.index}`,
      authenticate: true,
      response: text
    })
    return this
  }

  expectStatistics (statistics) {
    this.gets.push({
      path: `/statistics`,
      authenticate: false,
      response: statistics
    })
    return this
  }
}
