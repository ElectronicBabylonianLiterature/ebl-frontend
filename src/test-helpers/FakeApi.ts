import Promise from 'bluebird'
import Word from 'dictionary/domain/Word'

class Expectation {
  method: 'POST' | 'GET' = 'GET'
  path = ''
  authenticate = true
  response = {}
  verify = false
  called = false
  body: object | null = null
  isBlob = false

  constructor(data) {
    Object.assign(this, data)
  }
}

export default class FakeApi {
  private expectations: Expectation[] = []

  readonly client = {
    fetchJson: jest.fn().mockImplementation((path, authenticate) => {
      const expectation = this.expectations.find(
        entry =>
          entry.method === 'GET' &&
          entry.path === path &&
          entry.authenticate === authenticate &&
          !entry.isBlob
      )
      return expectation
        ? Promise.resolve(expectation.response)
        : Promise.reject(
            new Error(
              `Unexpected ${
                authenticate ? 'authenticated' : 'not-authenticated'
              } fetchJson: ${path}`
            )
          )
    }),

    postJson: jest.fn().mockImplementation(path => {
      const expectation = this.expectations.find(
        entry => entry.method === 'POST' && entry.path === path && !entry.isBlob
      )
      return expectation
        ? Promise.resolve(expectation.response)
        : Promise.reject(new Error(`Unexpected postJson: ${path}`))
    }),

    fetchBlob: jest.fn().mockImplementation((path, authenticate) => {
      const expectation = this.expectations.find(
        entry =>
          entry.method === 'GET' &&
          entry.path === path &&
          entry.authenticate === authenticate &&
          entry.isBlob
      )
      return expectation
        ? Promise.resolve(expectation.response)
        : Promise.reject(
            new Error(
              `Unexpected ${
                authenticate ? 'authenticated' : 'not-authenticated'
              } fetchBlob: ${path}`
            )
          )
    })
  }

  expectTexts(texts): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'GET',
        path: `/texts`,
        authenticate: false,
        response: texts,
        verify: true
      })
    )
    return this
  }

  allowText(text): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'GET',
        path: `/texts/${text.category}/${text.index}`,
        authenticate: true,
        response: text
      })
    )
    return this
  }

  expectText(text): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'GET',
        path: `/texts/${text.category}/${text.index}`,
        authenticate: true,
        response: text,
        verify: true
      })
    )
    return this
  }

  expectUpdateManuscripts(text, chapterIndex, manuscripts): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'POST',
        path: `/texts/${text.category}/${text.index}/chapters/${chapterIndex}/manuscripts`,
        response: text,
        verify: true,
        body: manuscripts
      })
    )
    return this
  }

  expectUpdateLines(text, chapterIndex, lines): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'POST',
        path: `/texts/${text.category}/${text.index}/chapters/${chapterIndex}/lines`,
        response: text,
        verify: true,
        body: lines
      })
    )
    return this
  }

  expectAnnotations(
    number: string,
    annotationDtos: readonly object[]
  ): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'GET',
        path: `/fragments/${number}/annotations`,
        authenticate: true,
        response: { annotations: annotationDtos },
        verify: true
      })
    )
    return this
  }

  expectUpdateAnnotations(
    number: string,
    annotationDtos: readonly object[]
  ): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'POST',
        path: `/fragments/${number}/annotations`,
        body: {
          fragmentNumber: number,
          annotations: annotationDtos
        },
        authenticate: true,
        response: { annotations: annotationDtos },
        verify: true
      })
    )
    return this
  }

  expectFragment(fragmentDto): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'GET',
        path: `/fragments/${fragmentDto._id}`,
        authenticate: true,
        response: fragmentDto,
        verify: true
      })
    )
    return this
  }

  expectPhoto(number: string): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'GET',
        path: `/fragments/${number}/photo`,
        authenticate: true,
        isBlob: true,
        verify: true
      })
    )
    return this
  }

  expectSearchWords(query: string, words: readonly Word[]): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'GET',
        path: `/words?query=${encodeURIComponent(query)}`,
        response: words,
        authenticate: true,
        verify: true
      })
    )
    return this
  }

  expectWord(word: Word): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'GET',
        path: `/words/${encodeURIComponent(word._id)}`,
        response: word,
        authenticate: true,
        verify: true
      })
    )
    return this
  }

  expectUpdateWord(word: Word): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'POST',
        path: `/words/${encodeURIComponent(word._id)}`,
        body: word,
        response: word,
        authenticate: true,
        verify: true
      })
    )
    return this
  }

  allowStatistics(statistics): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'GET',
        path: `/statistics`,
        authenticate: false,
        response: statistics
      })
    )
    return this
  }

  allowImage(file): FakeApi {
    this.expectations.push(
      new Expectation({
        method: 'GET',
        path: `/images/${file}`,
        authenticate: false,
        isBlob: true
      })
    )
    return this
  }

  verifyExpectations(): void {
    const methods = {
      GET: (expectation: Expectation): void =>
        expect(
          expectation.isBlob ? this.client.fetchBlob : this.client.fetchJson
        ).toHaveBeenCalledWith(expectation.path, expectation.authenticate),
      POST: (expectation: Expectation): void =>
        expect(this.client.postJson).toHaveBeenCalledWith(
          expectation.path,
          expectation.body || expect.anything()
        )
    }
    this.expectations
      .filter(expectation => expectation.verify)
      .forEach(expectation => methods[expectation.method](expectation))
  }
}
