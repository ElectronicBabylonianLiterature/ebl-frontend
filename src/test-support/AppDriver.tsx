import React from 'react'
import {
  fireEvent,
  render,
  RenderResult,
  act,
  Matcher,
  within,
  waitFor,
} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import _ from 'lodash'
import App from 'App'
import WordRepository from 'dictionary/infrastructure/WordRepository'
import FragmentRepository from 'fragmentarium/infrastructure/FragmentRepository'
import ApiImageRepository from 'fragmentarium/infrastructure/ImageRepository'
import FragmentService from 'fragmentarium/application/FragmentService'
import WordService from 'dictionary/application/WordService'
import TextService from 'corpus/application/TextService'
import MemorySession, { Session, guestSession } from 'auth/Session'
import BibliographyRepository from 'bibliography/infrastructure/BibliographyRepository'
import BibliographyService from 'bibliography/application/BibliographyService'
import FragmentSearchService from 'fragmentarium/application/FragmentSearchService'
import { Promise } from 'bluebird'
import { submitForm } from 'test-support/utils'
import { eblNameProperty, AuthenticationContext } from 'auth/Auth'
import SignRepository from 'signs/infrastructure/SignRepository'
import SignService from 'signs/application/SignService'

function createApp(api): JSX.Element {
  const wordRepository = new WordRepository(api)
  const fragmentRepository = new FragmentRepository(api)
  const imageRepository = new ApiImageRepository(api)
  const bibliographyRepository = new BibliographyRepository(api)
  const wordService = new WordService(wordRepository)
  const bibliographyService = new BibliographyService(bibliographyRepository)
  const fragmentService = new FragmentService(
    fragmentRepository,
    imageRepository,
    wordRepository,
    bibliographyService
  )
  const fragmentSearchService = new FragmentSearchService(fragmentRepository)
  const textService = new TextService(api, fragmentService, wordService)
  const signsRepository = new SignRepository(api)
  const signService = new SignService(signsRepository)
  return (
    <App
      signService={signService}
      wordService={wordService}
      fragmentService={fragmentService}
      fragmentSearchService={fragmentSearchService}
      bibliographyService={bibliographyService}
      textService={textService}
    />
  )
}

export default class AppDriver {
  private readonly api
  private initialEntries: string[] = []
  private view: RenderResult | null = null
  private session: Session | null = null

  constructor(api) {
    this.api = api
  }

  getView(): RenderResult {
    if (this.view) {
      return this.view
    } else {
      throw new Error('getElement called before render.')
    }
  }

  withPath(path: string): AppDriver {
    this.initialEntries = [path]
    return this
  }

  withSession(): AppDriver {
    this.session = new MemorySession([
      'write:texts',
      'read:fragments',
      'annotate:fragments',
      'read:words',
    ])
    return this
  }

  async render(): Promise<AppDriver> {
    await act(async () => {
      this.view = render(
        <MemoryRouter initialEntries={this.initialEntries}>
          <AuthenticationContext.Provider
            value={{
              login: _.noop,
              logout: _.noop,
              getSession: (): Session => this.session ?? guestSession,
              isAuthenticated: (): boolean => this.session !== null,
              getAccessToken(): Promise<string> {
                throw new Error('Not implemented')
              },
              getUser(): { [eblNameProperty]: string } {
                return { [eblNameProperty]: 'Test' }
              },
            }}
          >
            {createApp(this.api)}
          </AuthenticationContext.Provider>
        </MemoryRouter>
      )
    })

    return this
  }

  async waitForText(text: Matcher): Promise<void> {
    await this.getView().findByText(text)
  }

  async waitForTextToDisappear(text: Matcher): Promise<void> {
    await waitFor(() => {
      expect(this.getView().queryByText(text)).not.toBeInTheDocument()
    })
  }

  expectTextContent(text: string | RegExp): void {
    expect(this.getView().container).toHaveTextContent(text)
  }

  expectNotInContent(text: Matcher): void {
    expect(this.getView().queryByText(text)).not.toBeInTheDocument()
  }

  expectLink(text: Matcher, expectedHref: string): void {
    expect(this.getView().getByText(text)).toHaveAttribute('href', expectedHref)
  }

  expectBreadcrumbs(crumbs: readonly string[]): void {
    this.expectTextContent(crumbs.join(''))
  }

  expectBreadcrumb(crumb: string, link: string): void {
    expect(
      within(this.getView().getByLabelText('breadcrumb')).getByText(crumb)
    ).toHaveAttribute('href', link)
  }

  expectInputElement(label: Matcher, expectedValue: unknown): void {
    expect(this.getView().getByLabelText(label)).toHaveValue(
      String(expectedValue)
    )
  }

  expectChecked(label: Matcher): void {
    expect(this.getView().getByLabelText(label)).toBeChecked()
  }

  expectNotChecked(label: Matcher): void {
    expect(this.getView().getByLabelText(label)).not.toBeChecked()
  }

  async changeValueByLabel(label: Matcher, newValue: unknown): Promise<void> {
    const input = this.getView().getByLabelText(label)
    await act(async () => {
      fireEvent.change(input, { target: { value: newValue } })
    })
  }

  async submitForm(): Promise<void> {
    await submitForm(this.getView().container)
  }

  async click(text: Matcher, n = 0): Promise<void> {
    const clickable = this.getView().getAllByText(text)[n]
    await act(async () => {
      fireEvent.click(clickable)
    })
  }
}
