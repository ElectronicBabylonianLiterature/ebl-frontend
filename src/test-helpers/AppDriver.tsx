import React from 'react'
import {
  fireEvent,
  render,
  RenderResult,
  act,
  Matcher,
  within
} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import _ from 'lodash'
import App from 'App'
import WordRepository from 'dictionary/infrastructure/WordRepository'
import FragmentRepository from 'fragmentarium/infrastructure/FragmentRepository'
import ApiImageRepository from 'fragmentarium/infrastructure/ImageRepository'
import FragmentService from 'fragmentarium/application/FragmentService'
import WordService from 'dictionary/application/WordService'
import TextService from 'corpus/TextService'
import MemorySession, { Session, guestSession } from 'auth/Session'
import BibliographyRepository from 'bibliography/infrastructure/BibliographyRepository'
import BibliographyService from 'bibliography/application/BibliographyService'
import FragmentSearchService from 'fragmentarium/application/FragmentSearchService'
import { Promise } from 'bluebird'
import { submitForm } from 'test-helpers/utils'
import { eblNameProperty, AuthenticationContext } from 'auth/Auth'

function createApp(api): JSX.Element {
  const wordRepository = new WordRepository(api)
  const fragmentRepository = new FragmentRepository(api)
  const imageRepository = new ApiImageRepository(api)
  const bibliographyRepository = new BibliographyRepository(api)
  const wordService = new WordService(wordRepository)
  const bibliographyService = new BibliographyService(bibliographyRepository)
  const fragmentService = new FragmentService(
    fragmentRepository,
    imageRepository,
    wordRepository,
    bibliographyService
  )
  const fragmentSearchService = new FragmentSearchService(fragmentRepository)
  const textService = new TextService(api)

  return (
    <App
      wordService={wordService}
      fragmentService={fragmentService}
      fragmentSearchService={fragmentSearchService}
      bibliographyService={bibliographyService}
      textService={textService}
    />
  )
}

export default class AppDriver {
  private readonly api
  private initialEntries: string[] = []
  private element: RenderResult | null = null
  private session: Session | null = null

  constructor(api) {
    this.api = api
  }

  getElement(): RenderResult {
    if (this.element) {
      return this.element
    } else {
      throw new Error('getElement called before render.')
    }
  }

  withPath(path): AppDriver {
    this.initialEntries = [path]
    return this
  }

  withSession(): AppDriver {
    this.session = new MemorySession([
      'write:texts',
      'read:fragments',
      'annotate:fragments',
      'read:words'
    ])
    return this
  }

  async render(): Promise<AppDriver> {
    await act(async () => {
      this.element = render(
        <MemoryRouter initialEntries={this.initialEntries}>
          <AuthenticationContext.Provider
            value={{
              login: _.noop,
              logout: _.noop,
              getSession: (): Session => this.session ?? guestSession,
              isAuthenticated: (): boolean => this.session !== null,
              getAccessToken(): Promise<string> {
                throw new Error('Not implemented')
              },
              getUser(): { [eblNameProperty]: string } {
                return { [eblNameProperty]: 'Test' }
              }
            }}
          >
            {createApp(this.api)}
          </AuthenticationContext.Provider>
        </MemoryRouter>
      )
    })

    return this
  }

  async waitForText(text: Matcher): Promise<void> {
    await this.getElement().findByText(text)
  }

  expectTextContent(text: string | RegExp): void {
    expect(this.getElement().container).toHaveTextContent(text)
  }

  expectNotInContent(text: Matcher): void {
    expect(this.getElement().queryByText(text)).toBeNull()
  }

  expectLink(text: Matcher, expectedHref: string): void {
    expect(this.getElement().getByText(text)).toHaveAttribute(
      'href',
      expectedHref
    )
  }

  expectBreadcrumbs(crumbs: readonly string[]): void {
    this.expectTextContent(crumbs.join(''))
  }

  expectBreadcrumb(crumb: string, link: string): void {
    expect(
      within(this.getElement().getByLabelText('breadcrumb')).getByText(crumb)
    ).toHaveAttribute('href', link)
  }

  expectInputElement(label: Matcher, expectedValue: unknown): void {
    expect(
      (this.getElement().getByLabelText(label) as HTMLInputElement).value
    ).toEqual(String(expectedValue))
  }

  changeValueByLabel(label: Matcher, newValue): void {
    const input = this.getElement().getByLabelText(label)
    act(() => {
      fireEvent.change(input, { target: { value: newValue } })
    })
  }

  async submitForm(): Promise<void> {
    await submitForm(this.getElement())
  }

  async click(text: Matcher, n = 0): Promise<void> {
    const clickable = this.getElement().getAllByText(text)[n]
    await act(async () => {
      fireEvent.click(clickable)
    })
  }
}
