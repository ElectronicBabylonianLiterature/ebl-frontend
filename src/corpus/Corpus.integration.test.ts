import _ from 'lodash'
import AppDriver from 'test-helpers/AppDriver'
import FakeApi from 'test-helpers/FakeApi'

const textsDto = [
  {
    category: 1,
    index: 2,
    name: 'First Category 2',
    numberOfVerses: 99,
    approximateVerses: false,
    chapters: [],
  },
  {
    category: 2,
    index: 1,
    name: 'Second Category',
    numberOfVerses: 2,
    approximateVerses: false,
    chapters: [],
  },
  {
    category: 3,
    index: 1,
    name: 'Third Category',
    numberOfVerses: 45,
    approximateVerses: true,
    chapters: [],
  },
  {
    category: 1,
    index: 1,
    name: 'First Category 1',
    numberOfVerses: 99,
    approximateVerses: false,
    chapters: [],
  },
  {
    category: 4,
    index: 1,
    name: 'Hidden Text',
    numberOfVerses: 1,
    approximateVerses: false,
    chapters: [],
  },
]

let fakeApi: FakeApi
let appDriver: AppDriver

afterEach(() => {
  fakeApi.verifyExpectations()
})

beforeEach(() => {
  fakeApi = new FakeApi()
  fakeApi.expectTexts(textsDto)
  fakeApi.allowImage('LibraryCropped.svg')
  appDriver = new AppDriver(fakeApi.client).withPath('/corpus')
})

test('With session', async () => {
  await appDriver.withSession().render()
  await appDriver.waitForText(RegExp(_.escapeRegExp('Narrative Poetry')))
  expect(appDriver.getElement().container).toMatchSnapshot()
})

test('Without session', async () => {
  await appDriver.render()
  await appDriver.waitForText(RegExp(_.escapeRegExp('Narrative Poetry')))
  expect(appDriver.getElement().container).toMatchSnapshot()
})
