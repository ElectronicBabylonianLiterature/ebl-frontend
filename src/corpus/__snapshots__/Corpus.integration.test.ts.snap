// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With session 1`] = `
<div>
  <header
    class="Header"
  >
    <nav
      class="navbar navbar-expand-md navbar-light"
    >
      <div
        class="container"
      >
        <a
          class="active navbar-brand"
          href="/"
          title="electronic Babylonian Literature (eBL)"
        >
          <h1
            class="Header__title"
          >
            <span
              class="Header__title-main"
            >
              electronic
              <br />
              Babylonian
              <br />
              Literature
            </span>
            <small
              class="Header__title-abbreviation"
            >
              eBL
            </small>
          </h1>
        </a>
        <div
          class="navbar-collapse collapse"
          id="Header-1"
        >
          <div
            class="mx-auto navbar-nav"
          >
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/dictionary"
                href="/dictionary"
              >
                Dictionary
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="active nav-link"
                data-rb-event-key="/corpus"
                href="/corpus"
              >
                Corpus
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/fragmentarium"
                href="/fragmentarium"
              >
                Fragmentarium
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/bibliography"
                href="/bibliography"
              >
                Bibliography
              </a>
            </div>
          </div>
          <span
            class="navbar-text"
          >
            <button
              class="btn btn-outline-secondary btn-sm"
              type="button"
            >
              Logout 
              Test
            </button>
          </span>
        </div>
        <button
          aria-controls="Header-1"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
      </div>
    </nav>
  </header>
  <section
    class="App-content"
  >
    <header
      class="App-content__header"
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          class="breadcrumb"
        >
          <li
            class="breadcrumb-item active"
          >
            <a
              href="/"
            >
              eBL
            </a>
          </li>
          <li
            aria-current="page"
            class="breadcrumb-item active"
          >
            Corpus
          </li>
        </ol>
      </nav>
      <div
        class="float-right btn-toolbar"
        role="toolbar"
      />
      <h2>
        Corpus
      </h2>
    </header>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-md-5"
        >
          <section>
            <h3 />
            <ol
              class="container-fluid"
            />
          </section>
          <section>
            <h3>
              I. Narrative Poetry
            </h3>
            <ol
              class="container-fluid"
            >
              <li
                class="row"
              >
                <div
                  class="col-md-8"
                >
                  <a
                    href="/corpus/1/1"
                  >
                    1
                    . 
                    First Category 1
                  </a>
                </div>
                <div
                  class="col-md-4"
                >
                  
                  99
                   vv.
                </div>
              </li>
              <li
                class="row"
              >
                <div
                  class="col-md-8"
                >
                  <a
                    href="/corpus/1/2"
                  >
                    2
                    . 
                    First Category 2
                  </a>
                </div>
                <div
                  class="col-md-4"
                >
                  
                  99
                   vv.
                </div>
              </li>
            </ol>
          </section>
          <section>
            <h3>
              II. Monologue and dialogue literature
            </h3>
            <ol
              class="container-fluid"
            >
              <li
                class="row"
              >
                <div
                  class="col-md-8"
                >
                  <a
                    href="/corpus/2/1"
                  >
                    1
                    . 
                    Second Category
                  </a>
                </div>
                <div
                  class="col-md-4"
                >
                  
                  2
                   vv.
                </div>
              </li>
            </ol>
          </section>
          <section>
            <h3>
              III. Literary Hymns and Prayers
            </h3>
            <ol
              class="container-fluid"
            >
              <li
                class="row"
              >
                <div
                  class="col-md-8"
                >
                  <a
                    href="/corpus/3/1"
                  >
                    1
                    . 
                    Third Category
                  </a>
                </div>
                <div
                  class="col-md-4"
                >
                  Â±
                  45
                   vv.
                </div>
              </li>
            </ol>
          </section>
        </div>
        <div
          class="col-md-7"
        >
          <img
            alt="LibraryCropped.svg"
            class="img-fluid"
            src="http://example.com/images/LibraryCropped.svg"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Without session 1`] = `
<div>
  <header
    class="Header"
  >
    <nav
      class="navbar navbar-expand-md navbar-light"
    >
      <div
        class="container"
      >
        <a
          class="active navbar-brand"
          href="/"
          title="electronic Babylonian Literature (eBL)"
        >
          <h1
            class="Header__title"
          >
            <span
              class="Header__title-main"
            >
              electronic
              <br />
              Babylonian
              <br />
              Literature
            </span>
            <small
              class="Header__title-abbreviation"
            >
              eBL
            </small>
          </h1>
        </a>
        <div
          class="navbar-collapse collapse"
          id="Header-2"
        >
          <div
            class="mx-auto navbar-nav"
          >
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/dictionary"
                href="/dictionary"
              >
                Dictionary
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="active nav-link"
                data-rb-event-key="/corpus"
                href="/corpus"
              >
                Corpus
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/fragmentarium"
                href="/fragmentarium"
              >
                Fragmentarium
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="/bibliography"
                href="/bibliography"
              >
                Bibliography
              </a>
            </div>
          </div>
          <span
            class="navbar-text"
          >
            <button
              class="btn btn-outline-secondary btn-sm"
              type="button"
            >
              Login
            </button>
          </span>
        </div>
        <button
          aria-controls="Header-2"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
      </div>
    </nav>
  </header>
  <section
    class="App-content"
  >
    <header
      class="App-content__header"
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          class="breadcrumb"
        >
          <li
            class="breadcrumb-item active"
          >
            <a
              href="/"
            >
              eBL
            </a>
          </li>
          <li
            aria-current="page"
            class="breadcrumb-item active"
          >
            Corpus
          </li>
        </ol>
      </nav>
      <div
        class="float-right btn-toolbar"
        role="toolbar"
      />
      <h2>
        Corpus
      </h2>
    </header>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-md-5"
        >
          <section>
            <h3 />
            <ol
              class="container-fluid"
            />
          </section>
          <section>
            <h3>
              I. Narrative Poetry
            </h3>
            <ol
              class="container-fluid"
            >
              <li
                class="row"
              >
                <div
                  class="col-md-8"
                >
                  1
                  . 
                  First Category 1
                </div>
                <div
                  class="col-md-4"
                >
                  
                  99
                   vv.
                </div>
              </li>
              <li
                class="row"
              >
                <div
                  class="col-md-8"
                >
                  2
                  . 
                  First Category 2
                </div>
                <div
                  class="col-md-4"
                >
                  
                  99
                   vv.
                </div>
              </li>
            </ol>
          </section>
          <section>
            <h3>
              II. Monologue and dialogue literature
            </h3>
            <ol
              class="container-fluid"
            >
              <li
                class="row"
              >
                <div
                  class="col-md-8"
                >
                  1
                  . 
                  Second Category
                </div>
                <div
                  class="col-md-4"
                >
                  
                  2
                   vv.
                </div>
              </li>
            </ol>
          </section>
          <section>
            <h3>
              III. Literary Hymns and Prayers
            </h3>
            <ol
              class="container-fluid"
            >
              <li
                class="row"
              >
                <div
                  class="col-md-8"
                >
                  1
                  . 
                  Third Category
                </div>
                <div
                  class="col-md-4"
                >
                  Â±
                  45
                   vv.
                </div>
              </li>
            </ol>
          </section>
        </div>
        <div
          class="col-md-7"
        >
          <img
            alt="LibraryCropped.svg"
            class="img-fluid"
            src="http://example.com/images/LibraryCropped.svg"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
